### DISCUSSION TITLE
ScriptReference/Video.VideoPlayer-isPlaying

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#339] [Notes]
# ScriptReference/Video.VideoPlayer-isPlaying

https://docs.unity3d.com/ScriptReference/Video.VideoPlayer-isPlaying.html

**Comment by docscomments** [comment]
**Comment by `@mariosubspace`:**  

Doesn't seem to work even when calling `Prepare()` beforehand. More accurate to check if the frame is past 0 or other small number.

**Reply by kdchabuk** [reply]
`Prepare()` runs an asynchronous operation, so the video may not be ready to play immediately afterwards. Check `isPrepared` or use `prepareCompleted`.

**Comment by kdchabuk** [comment]
`isPlaying` is not guaranteed to become `false` at the end of the video. `Stop` or `Pause` are reliable ways to stop playback.

If `isLooping` is `true`, `loopPointReached` will be invoked at the end of the video, but this is not guaranteed when it is `false`.
